### PR TITLE
refactor: use build tags to disable, instead of to enable a feature

### PR DESCRIPTION
**Pull request by dunglas**
Also, add a way to fetch the latest C header provided by watcher using `go generate`.

### CODE REVIEW

**Comment by AlliBalliBaba** [comment]
Fetching the watcher header is probably a good idea. Would it make senso to add a check in the pipeline that will make the watcher installation fail if a new header is available?

**Comment by AlliBalliBaba** [comment]
Also msan fails again with the watcher. We can either disable tests or try our luck at a memory sanitized stdc++. I found this Docker image with instructions I haven't tried yet: https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/master/.github/msan-tester.Dockerfile

File: internal/watcher/watcher.go [hunk 1]
```diff
@@ -1,4 +1,6 @@
-//go:build watcher
+//go:build !nowatcher
+
+//go:generate curl --silent https://raw.githubusercontent.com/e-dant/watcher/refs/heads/release/watcher-c/include/wtr/watcher-c.h -o watcher-c.h
```
Comment (withinboredom) [inline internal/watcher/watcher.go:3]: This makes me want to create a go package template that just tracks releases and does a `go generate` to get cgo deps (like this one). Then all the package does is allow you to manage dependencies via normal ole' go.mod files.

**Review by AlliBalliBaba — APPROVED** [review]
Looks good 👍

**Review by withinboredom — APPROVED** [review]

File: internal/watcher/watcher.go (hunk 1 above)
Comment (dunglas) [reply internal/watcher/watcher.go:3]: We have a better option now thanks to @e-dant, I'm on it! https://github.com/e-dant/watcher/issues/58

**Pull request merged**